elastic/eui
# [Emotion] Convert EuiFilterGroup and EuiFilterButton

## Summary

- See https://github.com/elastic/eui/issues/6398
  - There's still one unconverted component before this issue closes (`EuiFilterSelectItem`), but this PR is getting pretty long as it is and already has its own set of Kibana changes, so I'll convert that component later separately.

I recommend [following along by commit](https://github.com/elastic/eui/pull/6957/commits) - as always, there's a good amount of incidental cleanup happening, so I tried to make individual commits as atomic and 1:1 as possible for easier review.

Other than snapshots, there is one active Kibana cleanup item that needs to be done as part of the next upgrade, which is converting all direct usages of `.euiFilterGroup__popoverPanel` to Emotion (see [`7412d84`](https://github.com/elastic/eui/commit/7412d848dbf03cce63d32d3019c93340dd355334?w=1)). @tkajtoch, let me know once you've created a Kibana upgrade PR and I'll push up the work needed for this to your branch.

## QA

- [ ] Compare [staging](https://eui.elastic.co/pr_6957/#/forms/filter-group) against [prod](https://elastic.github.io/eui/#/forms/filter-group) and confirm docs examples look the same as before
- `gh pr checkout 6957 && yarn storybook`
- Go to http://localhost:6006/?path=/story/euifiltergroup--playground and toggle various props, and confirm that nothing looks broken in any permutation
- Go to http://localhost:6006/?path=/story/euifilterbutton--playground and click around, and confirm that nothing looks broken in any permutation

### General checklist


### Emotion checklist

**General**

- ~[ ] Checked component playground~ - No playground exists for this component

---

**Unit tests**

- [x] Converted tests to RTL

---

**Sass/Emotion conversion process**

- ~[ ] Converted all global Sass vars/mixins to JS (e.g. `$euiSize` to `euiTheme.size.base`)~ No Sass vars
- [x] Removed converted Sass files

---

**CSS tech debt**

- ~[ ] Wrapped all animations or transitions in `euiCanAnimate`~ No animations

---

**DOM Cleanup**

    - `-` (single hyphen instead of double hyphen) modifiers left alone, as there's multiple references to them across Kibana. Double hyphen modifiers are in snapshots only and were removed

---

**Kibana due diligence**

   - ⚠️ KIBANA TODO: Direct references to `.euiFilterGroup__popoverPanel` need to be converted to Emotion instead (see 7412d848dbf03cce63d32d3019c93340dd355334)

---

**Extras/nice-to-have**
